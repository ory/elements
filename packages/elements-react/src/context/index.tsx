// Copyright Â© 2024 Ory Corp
// SPDX-License-Identifier: Apache-2.0

export { useComponents, useNodeSorter } from "./component"
export {
  useOryFlow,
  type FlowContextValue,
  type FlowContainerSetter,
} from "./flow-context"
export * from "./provider"
