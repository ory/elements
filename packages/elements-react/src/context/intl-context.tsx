// Copyright Â© 2024 Ory Corp
// SPDX-License-Identifier: Apache-2.0

import { merge } from "lodash"
import { PropsWithChildren } from "react"
import { IntlProvider as OriginalIntlProvider } from "react-intl"
import { LocaleMap, locales } from "../locales"

// ISO 639-1 language codes
// https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
export const LanguageCodes = [
  "ab",
  "aa",
  "af",
  "sq",
  "am",
  "ar",
  "hy",
  "as",
  "ay",
  "az",
  "ba",
  "eu",
  "bn",
  "dz",
  "bh",
  "bi",
  "br",
  "bg",
  "my",
  "be",
  "km",
  "ca",
  "zh",
  "co",
  "hr",
  "cs",
  "da",
  "nl",
  "en",
  "eo",
  "et",
  "fo",
  "fj",
  "fi",
  "fr",
  "fy",
  "gd",
  "gl",
  "ka",
  "de",
  "el",
  "kl",
  "gn",
  "gu",
  "ha",
  "iw",
  "hi",
  "hu",
  "is",
  "in",
  "ia",
  "ie",
  "ik",
  "ga",
  "it",
  "ja",
  "jw",
  "kn",
  "ks",
  "kk",
  "rw",
  "ky",
  "rn",
  "ko",
  "ku",
  "lo",
  "la",
  "lv",
  "ln",
  "lt",
  "mk",
  "mg",
  "ms",
  "ml",
  "mt",
  "mi",
  "mr",
  "mo",
  "mn",
  "na",
  "ne",
  "no",
  "oc",
  "or",
  "om",
  "ps",
  "fa",
  "pl",
  "pt",
  "pa",
  "qu",
  "rm",
  "ro",
  "ru",
  "sm",
  "sg",
  "sa",
  "sr",
  "sh",
  "st",
  "tn",
  "sn",
  "sd",
  "si",
  "ss",
  "sk",
  "sl",
  "so",
  "es",
  "su",
  "sw",
  "sv",
  "tl",
  "tg",
  "ta",
  "tt",
  "te",
  "th",
  "bo",
  "ti",
  "to",
  "ts",
  "tr",
  "tk",
  "tw",
  "uk",
  "ur",
  "uz",
  "vi",
  "vo",
  "cy",
  "wo",
  "xh",
  "ji",
  "yo",
  "zu",
] as const

export type Locale = keyof typeof locales

export type IntlContextProps = {
  locale: Locale
  customTranslations?: Partial<LocaleMap>
}

export const IntlProvider = ({
  children,
  locale,
  customTranslations,
}: PropsWithChildren<IntlContextProps>) => {
  const messages = merge({}, locales, customTranslations)

  return (
    <OriginalIntlProvider
      onWarn={() => ({})}
      defaultRichTextElements={{
        del: (chunks) => <del>{chunks}</del>,
      }}
      locale={locale}
      messages={messages[locale]}
      defaultLocale="en"
    >
      {children}
    </OriginalIntlProvider>
  )
}
