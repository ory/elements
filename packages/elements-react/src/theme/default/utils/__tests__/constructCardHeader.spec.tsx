// Copyright Â© 2024 Ory Corp
// SPDX-License-Identifier: Apache-2.0

import { FlowType, UiNode, UiTextTypeEnum } from "@ory/client-fetch"
import { useCardHeaderText } from "../constructCardHeader"
import { renderHook } from "@testing-library/react"
import { PropsWithChildren } from "react"
import { IntlProvider } from "../../../../context/intl-context"

const password: UiNode = {
  group: "password",
  type: "text",
  attributes: {
    node_type: "input",
    name: "password",
    type: "password",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {},
}

const oidc: UiNode = {
  group: "oidc",
  type: "text",
  attributes: {
    node_type: "input",
    name: "oidc",
    type: "hidden",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {},
}

const code: UiNode = {
  group: "code",
  type: "text",
  attributes: {
    node_type: "input",
    name: "code",
    type: "text",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {},
}

const passkey: UiNode = {
  group: "passkey",
  type: "text",
  attributes: {
    node_type: "input",
    name: "passkey",
    type: "text",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {},
}

const webauthn: UiNode = {
  group: "webauthn",
  type: "text",
  attributes: {
    node_type: "input",
    name: "webauthn",
    type: "text",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {},
}

const defaultGroup: UiNode = {
  group: "default",
  type: "input",
  attributes: {
    node_type: "input",
    name: "traits.email",
    type: "text",
    value: "",
    disabled: false,
    label: {
      text: "Email",
      type: UiTextTypeEnum.Info,
      id: 9999,
    },
  },
  messages: [],
  meta: {
    label: {
      text: "Email",
      type: UiTextTypeEnum.Info,
      id: 9999,
    },
  },
}

const identifierFirst: UiNode = {
  group: "identifier_first",
  type: "input",
  attributes: {
    node_type: "input",
    name: "identifier_first",
    type: "text",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {
    label: {
      text: "Email",
      type: UiTextTypeEnum.Info,
      id: 9999,
    },
  },
}

const identifierFirstPhone: UiNode = {
  group: "identifier_first",
  type: "input",
  attributes: {
    node_type: "input",
    name: "identifier_first",
    type: "text",
    value: "",
    disabled: false,
  },
  messages: [],
  meta: {
    label: {
      text: "Phone number",
      type: UiTextTypeEnum.Info,
      id: 9999,
    },
  },
}

const combinations = {
  oidc,
  code,
  passkey,
  webauthn,
  defaultGroup,
  identifierFirst,
  password,
  identifierFirstPhone,
  default_and_password: [defaultGroup, password],
  default_and_code: [defaultGroup, code],
  default_and_oidc: [defaultGroup, oidc],
  default_and_passkeys: [defaultGroup, passkey],
  default_and_webauthn: [defaultGroup, webauthn],
  idenfier_first_and_oidc: [identifierFirst, oidc],
}

const wrapper = ({ children }: PropsWithChildren) => (
  <IntlProvider locale="en">{children}</IntlProvider>
)

for (const flowType of [
  FlowType.Login,
  FlowType.Registration,
  FlowType.Verification,
  FlowType.Recovery,
  FlowType.Settings,
]) {
  describe("flowType=" + flowType, () => {
    for (const refresh of [true, false]) {
      // Yes, it doesn't make sense to test other flows with refresh enabled, but it doesn't hurt, and typescript is dumb here.
      describe("refresh=" + refresh, () => {
        const opts = {
          flowType,
          flow: { refresh },
        }
        for (const [key, value] of Object.entries(combinations)) {
          describe("combination=" + key, () => {
            test("constructCardHeaderText", () => {
              const res = renderHook(
                () =>
                  useCardHeaderText(
                    Array.isArray(value) ? value : [value],
                    opts,
                  ),
                { wrapper },
              )
              expect(res.result.current).toMatchSnapshot()
            })
          })
        }
      })
    }
  })
}
