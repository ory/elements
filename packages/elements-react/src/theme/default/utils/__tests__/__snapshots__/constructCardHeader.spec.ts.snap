// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flowType=login refresh=false combination=code constructCardHeaderText 1`] = `
{
  "description": "Sign in with a code sent to your email.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Sign in with a code sent to your email.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign in with a social provider.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Sign in with a Passkey.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Sign in with your email and password.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign in with a security key.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign in with a social provider or your undefined.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Sign in with your undefined.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Sign in with your undefined.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Sign in with a social provider.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Sign in with a Passkey.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=password constructCardHeaderText 1`] = `
{
  "description": "Sign in with your email and password.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=false combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign in with a security key.",
  "title": "Sign in",
}
`;

exports[`flowType=login refresh=true combination=code constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a code sent to your email.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a code sent to your email.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a social provider.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a Passkey.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with your email and password.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a security key.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with .",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a social provider or your undefined.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with your undefined.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with your undefined.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a social provider.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a Passkey.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=password constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with your email and password.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=login refresh=true combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Confirm your identity with a security key.",
  "title": "Reauthenticate",
}
`;

exports[`flowType=recovery refresh=false combination=code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=false combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=recovery refresh=true combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to receive a one-time access code",
  "title": "Recover your account",
}
`;

exports[`flowType=registration refresh=false combination=code constructCardHeaderText 1`] = `
{
  "description": "Sign up with a code sent to your email.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Sign up with a code sent to your email.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Sign up with a Passkey.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Sign up with your email and a password.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign up with a security key.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider or your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Sign up with your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Sign up with your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Sign up with a Passkey.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=password constructCardHeaderText 1`] = `
{
  "description": "Sign up with your email and a password.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=false combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign up with a security key.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=code constructCardHeaderText 1`] = `
{
  "description": "Sign up with a code sent to your email.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Sign up with a code sent to your email.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Sign up with a Passkey.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Sign up with your email and a password.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign up with a security key.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider or your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Sign up with your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Sign up with your undefined.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Sign up with a social provider.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Sign up with a Passkey.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=password constructCardHeaderText 1`] = `
{
  "description": "Sign up with your email and a password.",
  "title": "Sign up",
}
`;

exports[`flowType=registration refresh=true combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Sign up with a security key.",
  "title": "Sign up",
}
`;

exports[`flowType=settings refresh=false combination=code constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=password constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=false combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=code constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=password constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=settings refresh=true combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Update your account settings",
  "title": "Update your account",
}
`;

exports[`flowType=verification refresh=false combination=code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=false combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=default_and_code constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=default_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=default_and_passkeys constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=default_and_password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=default_and_webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=defaultGroup constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=idenfier_first_and_oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=identifierFirst constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=identifierFirstPhone constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=oidc constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=passkey constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=password constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;

exports[`flowType=verification refresh=true combination=webauthn constructCardHeaderText 1`] = `
{
  "description": "Enter the email address associated with your account to verify it",
  "title": "Verify your account",
}
`;
